### PR TITLE
raftstore/pd: replace FutureWorker with Worker for pd worker

### PR DESCRIPTION
**Pull request by 5kbpers**
### What problem does this PR solve?

Problem Summary:

For https://github.com/tikv/tikv/issues/10539 we need to implement a tick for pd worker to check the status of tikv. But `FutureWorker` doesn't support tick.

### What is changed and how it works?

Replace `FutureWorker` with `Worker` for pd worker.

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
N/A
```

### CODE REVIEW

**Comment by CodingPoeta** [comment]
lalaland

**Comment by CodingPoeta** [comment]
hey

**Comment by CodingPoeta** [comment]
hahaha

**Comment by CodingPoeta** [comment]
lalaland

**Comment by CodingPoeta** [comment]
test

**Comment by CodingPoeta** [comment]
life is a drug

**Comment by CodingPoeta** [comment]
life is a drug

**Comment by CodingPoeta** [comment]
truetrue

File: components/raftstore/src/store/fsm/store.rs [hunk 1]
```diff
@@ -1408,9 +1409,6 @@ impl<EK: KvEngine, ER: RaftEngine> RaftBatchSystem<EK, ER> {
         MEMTRACE_RAFT_ROUTER_ALIVE.trace(TraceEvent::Reset(0));
         MEMTRACE_RAFT_ROUTER_LEAK.trace(TraceEvent::Reset(0));
 
-        if let Some(h) = handle {
```
Comment (sticnarf) [inline components/raftstore/src/store/fsm/store.rs:1411]: Does this change the behavior of `shutdown`?

**Review by BusyJay — COMMENTED** [review]
As far as I know, Lazy worker shares a thread pool with other workers, and we have seem an overloaded worker cause starvation on other workers and lead to deadlock in some case. Is it safe for PD worker?

**Comment by 5kbpers** [comment]
> As far as I know, Lazy worker shares a thread pool with other workers, and we have seem an overloaded worker cause starvation on other workers and lead to deadlock in some case. Is it safe for PD worker?

It will create a new thread pool, lazy worker here is just for creating the runner laziliy.

File: components/raftstore/src/store/fsm/store.rs (hunk 1 above)
Comment (5kbpers) [reply components/raftstore/src/store/fsm/store.rs:1411]: It does change the shutdown of pd worker from asynchronously to synchronously, but I think this doesn't have other side effects.

**Review by BusyJay — APPROVED** [review]

**Review by Little-Wallace — APPROVED** [review]
LGTM. Maybe you shall run it in a real TiKV cluster to avoid some problems we do not find.

**Comment by 5kbpers** [comment]
> LGTM. Maybe you shall run it in a real TiKV cluster to avoid some problems we do not find.

Thanks! Have been running for a few days.

**Pull request merged**